### PR TITLE
chore: upgrade rrweb for playback

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Connected to https://github.com/PostHog/posthog-js/pull/1115

## Changes

- Upgrade rrweb
- Reapply patches from https://github.com/PostHog/posthog/commits/8edc743a7d52cd5ff118fa2c3a773f79f512f69b/patches/rrweb%402.0.0-alpha.11.patch

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Ensured replays (including canvas playback) still works

### CODE REVIEW

File: patches/rrweb@2.0.0-alpha.12.patch [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+--- a/es/rrweb/packages/rrweb/src/replay/index.js
++++ b/es/rrweb/packages/rrweb/src/replay/index.js
+@@ -201,6 +201,10 @@ class Replayer {
+             mouseTail: defaultMouseTailConfig,
+             useVirtualDom: true,
+             logger: console,
++            onError: (e) => {
++                // maintain the existing behaviour of throwing if no handler is provided
++                throw e;
++            },
```
Comment (daibhin) [inline patches/rrweb@2.0.0-alpha.12.patch:12]: Very difficult to parse this patch but it's just this change and adding a try / catch

File: patches/rrweb@2.0.0-alpha.12.patch [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+-            case IncrementalSource.AdoptedStyleSheet: {
+-                if (this.usingVirtualDom)
+-                    this.adoptedStyleSheets.push(d);
+-                else
+-                    this.applyAdoptedStyleSheet(d);
+-                break;
+             }
++        } catch (error) {
++            this.config.onError(error);
+         }
```
Comment (daibhin) [inline patches/rrweb@2.0.0-alpha.12.patch:520]: End of the try / catch

File: patches/rrweb@2.0.0-alpha.12.patch [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+-            case IncrementalSource.MouseMove:
+-                if (isSync) {
+-                    const lastPosition = d.positions[d.positions.length - 1];
+-                    this.mousePos = {
+-                        x: lastPosition.x,
+-                        y: lastPosition.y,
+-                        id: lastPosition.id,
+-                        debugData: d,
+-                    };
++        try {
```
Comment (daibhin) [inline patches/rrweb@2.0.0-alpha.12.patch:41]: Start of try catch

File: ee/frontend/mobile-replay/schema/web/rr-web-schema.json [hunk 4]
```diff
@@ -631,7 +658,7 @@
                     "type": "number"
                 }
             },
-            "required": ["id", "source", "type", "x", "y"],
+            "required": ["id", "source", "type"],
```
Comment (marandaneto) [inline ee/frontend/mobile-replay/schema/web/rr-web-schema.json:661]: x and y aren't required anymore?

**Review by marandaneto — APPROVED** [review]

File: ee/frontend/mobile-replay/schema/web/rr-web-schema.json (hunk 4 above)
Comment (daibhin) [reply ee/frontend/mobile-replay/schema/web/rr-web-schema.json:661]: Exactly, comes from this change that was merged: https://github.com/rrweb-io/rrweb/pull/1375

**Pull request merged**